Mage.Sets/src/mage/cards/p/ProgenitorsIcon.java

```diff
@@ -0,0 +1,148 @@
+
+/**
+ *
+ * @author jimga150
+ */
+public final class ProgenitorsIcon extends CardImpl {
+
+    public ProgenitorsIcon(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.ARTIFACT}, "{3}");
+        
+
+        // As Progenitor's Icon enters the battlefield, choose a creature type.
+        // Based on Circle of Solace
+        this.addAbility(new AsEntersBattlefieldAbility(new ChooseCreatureTypeEffect(Outcome.Neutral)));
+
+        // {T}: Add one mana of any color.
+        this.addAbility(new AnyColorManaAbility());
+
+        // {T}: The next spell of the chosen type you cast this turn can be cast as though it had flash.
+        // Based on Ride the Avalanche
+        Ability ability = new SimpleActivatedAbility(Zone.BATTLEFIELD, new ProgenitorsIconAsThoughEffect(), new TapSourceCost());
+        ability.addWatcher(new ProgenitorsIconWatcher());
+        this.addAbility(ability);
+    }
+
+    private ProgenitorsIcon(final ProgenitorsIcon card) {
+        super(card);
+    }
+
+    @Override
+    public ProgenitorsIcon copy() {
+        return new ProgenitorsIcon(this);
+    }
+}
+
+class ProgenitorsIconAsThoughEffect extends AsThoughEffectImpl {
+
+    public ProgenitorsIconAsThoughEffect() {
+        super(AsThoughEffectType.CAST_AS_INSTANT, Duration.EndOfTurn, Outcome.Benefit);
+        staticText = "The next spell of the chosen type you cast this turn can be cast as though it had flash";
+    }
+
+    private ProgenitorsIconAsThoughEffect(final ProgenitorsIconAsThoughEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public void init(Ability source, Game game) {
+        SubType subType = ChooseCreatureTypeEffect.getChosenCreatureType(source.getSourceId(), game);
+        ProgenitorsIconWatcher.addPlayer(source.getControllerId(), subType, game);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        return true;
+    }
+
+    @Override
+    public ProgenitorsIconAsThoughEffect copy() {
+        return new ProgenitorsIconAsThoughEffect(this);
+    }
+
+    @Override
+    public boolean applies(UUID sourceId, Ability source, UUID affectedControllerId, Game game) {
+        if (!source.isControlledBy(affectedControllerId)) {
+            return false;
+        }
+        SubType subType = ChooseCreatureTypeEffect.getChosenCreatureType(source.getSourceId(), game);
+
+        if (!ProgenitorsIconWatcher.checkPlayerCast(affectedControllerId, subType, game)) {
+            discard();
+            return false;
+        }
+        Card card = game.getCard(sourceId);
+        if (card == null){
+            return false;
+        }
+
+        return card.getSubtype().contains(subType);
+    }
+}
+
+class ProgenitorsIconWatcher extends Watcher {
+
+    // Handles multiple instances of this same card choosing the same or different subtypes for the same player
+    private final Map<UUID, ArrayList<SubType>> playerSubTypesChosen = new HashMap<>();
```

Comment:
Is there a reason to use a List rather than a Set? Also, use the interface (List/Set) rather than an implementation (ArrayList/HashSet) when defining the type of a field or variable

Reply:
This is to handle cases where multiple creature types have been chosen for one player with multiple progenitor's icons.

Reply:
Yeah, that's why it's a Collection of multiple subtypes... but a Set doesn't have duplicates (while a List can) and I'm pretty sure that duplicates aren't relevant in this usage so a Set would be more appropriate?

Reply:
Hm.. what is the intended behavior if multiple progenitor's icons get played by the same player with the same creature subtype chosen, then are both tapped for their flash effect? Should multiple spells of that subtype be able to be played with flash after that, or should only one?

Reply:
Well, I'm not a judge, but my interpretation is there's only one "next" spell